### PR TITLE
Rule table merge

### PR DESCRIPTION
**Pull request by jworcestBAH**
This PR consolidates rules into a single Rule table, and removes separate handling for multi-field rules, which are now distinguished by a separate rule_timing_id.

### CODE REVIEW

**Comment by klrBAH** [comment]
I had to stop the tests because of a conflict on the jenkins box, so [run tests]

File: dataactvalidator/config/awardFinancialRules.csv [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+SUM_FIELDS,FALSE,,GrossOutlaysUndeliveredOrdersPrepaidTotal_CPE,"USSGL480200_UndeliveredOrdersObligationsPrepaidAdvanced_CPE, USSGL483200_UndeliveredOrdersObligationsTransferredPrepaidAdvanced_CPE, USSGL487200_DownwardAdjustmentsOfPriorYearPrepaidAdvancedUndeliveredOrdersObligationsRefundsCollected_CPE, USSGL488200_UpwardAdjustmentsOfPriorYearUndeliveredOrdersObligationsPrepaidAdvanced_CPE",GrossOutlaysUndeliveredOrdersPrepaidTotal (CPE) = USSGL(4802 + 4832 + 4872+ 4882). This applies to the award level.,4,C6B
+SUM_FIELDS,FALSE,,GrossOutlaysDeliveredOrdersPaidTotal_CPE,"USSGL490200_DeliveredOrdersObligationsPaid_CPE, USSGL490800_AuthorityOutlayedNotYetDisbursed_CPE, USSGL497200_DownwardAdjustmentsOfPriorYearPaidDeliveredOrdersObligationsRefundsCollected_CPE, USSGL498200_UpwardAdjustmentsOfPriorYearDeliveredOrdersObligationsPaid_CPE",GrossOutlaysDeliveredOrdersPaidTotal (CPE) = USSGL(4902 + 4908 + 4972 + 4982). This applies to the award level.,4,C7B
+REQUIRED_CONDITIONAL,TRUE,TransactionObligatedAmount,C17B,,Required on rows that do not include values for USSGL accounts balances or subtotals,1,C17
```
Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: Should there be another comma after TRUE (i.e., 2nd value in this line). Looks like a copy and pasted might have moved some commas around.

File: dataactvalidator/config/awardFinancialRules.csv [hunk 2]
```diff
@@ -1,16 +1,16 @@
 rule_type,is_single_field,field_name,rule_text_one,rule_text_two,description,rule_timing,rule_label
```
Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: The order of the `rule_timing` and `rule_label` in this file doesn't match the order in appropRules.csv.

File: dataactvalidator/config/awardFinancialRules.csv (hunk 1 above)
Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: James explained this...it's fine.

File: dataactvalidator/filestreaming/schemaLoader.py [hunk 3]
```diff
@@ -86,10 +91,10 @@ def loadCrossRules(filename):
                     targetFileId = validationDb.getFileId(record["target_file"])
                 else:
                     targetFileId = None
-                validationDb.addMultiFieldRule(
+                validationDb.addRule(
                     fileId, record["rule_type"], record["rule_text_one"],
-                    record["rule_text_two"], record["description"],
-                    record["rule_label"], record["rule_timing"], targetFileId)
+                    record["rule_text_two"], record["description"], record["rule_timing"],
+                    record["rule_label"], targetFileId, fileId = fileId)
```
Comment (bsweger) [inline dataactvalidator/filestreaming/schemaLoader.py]: Should the crossFile rule load have the same type of error message re: required fields as the regular rules load?

Comment (bsweger) [inline dataactvalidator/filestreaming/schemaLoader.py]: Will crossFile validations always have a target file? If so, should we check to make sure it's there before inserting the rule?

File: dataactvalidator/config/awardFinancialRules.csv (hunk 2 above)
Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: Even though this order doesn't affect the code, did a push that submits makes `awardFinancialRules.csv` consistent with the other rules files.

**Comment by bsweger** [comment]
@jworcestBAH Done reviewing. I did push a change to `awardFinancialRules.csv` so the order of those columns would match the other files. Can you take a quick look and give a 👍 if you're okay with the change?

File: dataactvalidator/config/awardFinancialRules.csv (hunk 2 above)
Comment (jworcestBAH) [inline dataactvalidator/config/awardFinancialRules.csv]: Thanks, agreed that it's best to keep it consistent.

File: dataactvalidator/filestreaming/schemaLoader.py (hunk 3 above)
Comment (jworcestBAH) [inline dataactvalidator/filestreaming/schemaLoader.py]: They do not currently always have a target file as I implemented the first rule types before you added the target file field.  Refactoring those to use target file would be worthwhile, it's on my list of refactors to do if I have time to spare at the end of this sprint.

**Comment by bsweger** [comment]
[run-tests]

**Comment by bsweger** [comment]
Ok, looks good!

**Pull request merged**